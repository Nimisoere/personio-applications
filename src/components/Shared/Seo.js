import React from "react";
import PropTypes from "prop-types";
import { Helmet } from "react-helmet";
import { appConstants } from "../../_constants";

export const Seo = ({ title, description, base }) => (
  <div>
    {base ? (
      <Helmet
        titleTemplate={`%s - ${title}`}
        defaultTitle={appConstants.appName}
      >
        <meta charSet="utf-8" />
        <title>
          {title} | {appConstants.appName}{" "}
        </title>
        <meta name="description" content={description} />
      </Helmet>
    ) : (
      <Helmet>
        <title>
          {title} | {appConstants.appName}{" "}
        </title>
        <meta name="description" content={description} />
      </Helmet>
    )}
  </div>
);

Seo.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  base: PropTypes.bool
};
