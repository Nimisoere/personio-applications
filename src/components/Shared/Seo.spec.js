import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {Seo} from "./Seo";

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    text: "test",
    blur: true
  };
  const enzymeWrapper = shallow(<Seo {...props} />);
  return {
    props,
    enzymeWrapper
  };
}

describe("components", () => {
  describe("SEO", () => {
    it("should render", () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
