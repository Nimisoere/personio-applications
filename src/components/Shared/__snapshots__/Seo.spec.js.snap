// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components SEO should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Seo
    blur={true}
    text="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <HelmetWrapper
        defer={true}
        encodeSpecialCharacters={true}
      >
        <title>
           | 
          Personio Applications
           
        </title>
        <meta
          name="description"
        />
      </HelmetWrapper>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <title>
             | 
            Personio Applications
             
          </title>,
          <meta
            name="description"
          />,
        ],
        "defer": true,
        "encodeSpecialCharacters": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " | ",
              "Personio Applications",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " | ",
            "Personio Applications",
            " ",
          ],
          "type": "title",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "content": undefined,
            "name": "description",
          },
          "ref": null,
          "rendered": null,
          "type": "meta",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
        >
          <title>
             | 
            Personio Applications
             
          </title>
          <meta
            name="description"
          />
        </HelmetWrapper>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <title>
               | 
              Personio Applications
               
            </title>,
            <meta
              name="description"
            />,
          ],
          "defer": true,
          "encodeSpecialCharacters": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " | ",
                "Personio Applications",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " | ",
              "Personio Applications",
              " ",
            ],
            "type": "title",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": undefined,
              "name": "description",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
